src/kernel.ts

```diff
@@ -884,12 +905,15 @@ class Kernel implements IKernel {
       return;
     }
     var content = msg.content as ICommOpen;
-    if (!content.target_module) {
-      this.commOpened.emit(msg);
+    this.commOpened.emit(msg);
+    var targetName = content.target_name;
+    var moduleName = content.target_module;
+    if (this._targetRegistry[targetName]) {
+      this._targetRegistry[targetName](msg);
+    }
```

Comment:
Okay, so Sylvain and I think it would be better to get rid of the registry, *always* use the static `(module, target)` based loaders, and eventually provide an easy way to hack into the general import/loader semantics to dynamically load code